libcontainer/process_linux.go

```diff
@@ -47,12 +48,20 @@ type parentProcess interface {
 	externalDescriptors() []string
 
 	setExternalDescriptors(fds []string)
+
+	forwardChildLogs()
+}
+
+type readWritePair struct {
+	r *os.File
+	w *os.File
 }
 
 type setnsProcess struct {
 	cmd             *exec.Cmd
 	parentPipe      *os.File
 	childPipe       *os.File
```

Comment:
Did you want to consolidate `parentPipe`/`childPipe` as well? Might not be a bad idea.

Reply:
We did it but renamed `readWritePair` to `filePair` since the init sock pair involves bidirectional communication so `readWritePair` was semantically incorrect.

Reply:
Seems reasonable.